diff -up xfig.3.2.5a/w_print.c~ xfig.3.2.5a/w_print.c
--- xfig.3.2.5a/w_print.c~	2009-04-08 16:44:24.000000000 +0200
+++ xfig.3.2.5a/w_print.c	2009-04-08 16:45:35.000000000 +0200
@@ -958,7 +958,7 @@ void create_print_panel(Widget w)
 	   var and put it into the widget */
 	if (emptyname(printer_val)) {
 		printer_val=getenv("PRINTER");
-		if (strchr(printer_val,'\\')) {
+		if (printer_val && strchr(printer_val,'\\')) {
 		    buf[0]='\0';
 		    len=0;
 		    for (i=0; i<strlen(printer_val); i++) {
